src/common.rs

```diff
@@ -4,6 +4,91 @@
 
 use arrayvec::ArrayVec;
 
+/// Defines a trait that chooses between libstd or libm implementations of float methods.
```

Comment:
so... is this actually necessary in the case of std? If std is present then these methods will already exist on our types, and method resolution will work, so it feels like we could get away with only defining this trait if the `musl` feature is active?

(the fact that this trait does nothing if `std` is active is supported by the need to add `#[allow(unused_code)]` on the import)